### PR TITLE
[REF] Rename `get_mask_volume`

### LINKED ISSUE #4639: Rename `get_mask_volume` to signify both -- calculation of volume for Nifti image and vertices for Surface image
```
    mask_n_vertices = get_mask_volume(mask)
```

we will have to rename this private function or split its behavior because something called get_mask_volume that returns a number of vertices is confusing

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/4630#discussion_r1799770140_

### PR DESCRIPTION
**Pull request by man-shu**
- Closes #4639

Changes proposed in this pull request:

- rename `get_mask_volume` to `get_mask_extent`
- update function refs, and related variables similarly

### CODE REVIEW

**Review by Remi-Gau — APPROVED** [review]
good with me

**Review by bthirion — COMMENTED** [review]
LGTM, but I think we should make the function "private"

File: nilearn/_utils/param_validation.py
Comment (Remi-Gau) [reply nilearn/_utils/param_validation.py:88]: it is in the `_utils` subpackage so it won't be part of the public API, but given that the function is only used in `nilearn/_utils/param_validation.py` (except for one test) we can still add a leading underscore to denote it is private to this module.

Comment (man-shu) [reply nilearn/_utils/param_validation.py:88]: So no underscore for this function?

Comment (Remi-Gau) [reply nilearn/_utils/param_validation.py:88]: nope

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/_utils/param_validation.py
Comment (man-shu) [reply nilearn/_utils/param_validation.py:88]: ok then adding the underscore

**Comment by man-shu** [comment]
Changed `get_mask_extent` -> `_get_mask_extent`.
Just waiting for the CI checks before merging...

**Comment by Remi-Gau** [comment]
Thanks @man-shu for taking care of this.

**Pull request merged**
